bokehjs/test/unit/models/annotations/box_annotation.ts

```diff
@@ -0,0 +1,121 @@
+    const {view} = await display(plot)
+    return view
+  }
+
+  function get_cursor(plot_view: PlotView): string {
+    return getComputedStyle(plot_view.canvas_view.events_el).cursor
+  }
+
+  async function get_cursors(ac: PlotActions, events: Iterable<UIEvent>): Promise<string[]> {
+    const cursors = []
+    for await (const _ of ac._emit(events)) {
+      cursors.push(get_cursor(ac.target))
+    }
+    return cursors
+  }
+
+  describe("when movable without handles", () => {
+    function mkbox() {
+      return new BoxAnnotation({
+        left: 0,
+        top: 0,
+        right: 0.5,
+        bottom: 0.5,
+        editable: true,
+        movable: "both",
+      })
+    }
+
+    it("should set cursor to \"grab\" inside box", async () => {
+      const plot_view = await mkplot(mkbox())
+      const ac = actions(plot_view, {units: "data"})
+
+      // inside box
+      await ac.hover(xy(0.25, 0.25))
+      expect(get_cursor(plot_view)).to.be.equal("grab")
+
+      // outside box
+      await ac.hover(xy(0.75, 0.75))
+      expect(get_cursor(plot_view)).to.be.equal("default")
+    })
+
+    it("should set cursor to \"grabbing\" when panning", async () => {
+      const plot_view = await mkplot(mkbox())
+      const ac = actions(plot_view, {units: "data"})
+
+      // First, hover to put the cursor in the correct initial state ("grab")
+      await ac.hover(xy(0.25, 0.25))
+
+      // Next, pan
+      const events = ac._pan({type: "line", xy0: xy(0.25, 0.25), xy1: xy(0.75, 0.75), n: 5})
+      const cursors = await get_cursors(ac, events)
+
+      expect([...no_repeated(cursors)]).to.be
+        .equal([...no_repeated(["grab", "grabbing", "grab"])])
+    })
+  })
+
+  describe("when movable with handles", () => {
+    function mkbox() {
+      return new BoxAnnotation({
+        left: 0,
+        top: 0,
+        right: 0.5,
+        bottom: 0.5,
+        editable: true,
+        movable: "both",
+        use_handles: true,
+      })
+    }
+
+    it("should set cursor to \"move\" in the middle", async () => {
+      const plot_view = await mkplot(mkbox())
+      const ac = actions(plot_view, {units: "data"})
+
+      // inside box
+      await ac.hover(xy(0.25, 0.25))
+      expect(get_cursor(plot_view)).to.be.equal("move")
+
+      // outside box
+      await ac.hover(xy(0.75, 0.75))
+      expect(get_cursor(plot_view)).to.be.equal("default")
+    })
+
+    it("should set cursor to \"move\" when panning", async () => {
+      const plot_view = await mkplot(mkbox())
+      const ac = actions(plot_view, {units: "data"})
+
+      // First, hover to put the cursor in the correct initial state ("move")
+      await ac.hover(xy(0.25, 0.25))
+
+      // Next, pan
+      const events = ac._pan({type: "line", xy0: xy(0.25, 0.25), xy1: xy(0.75, 0.75), n: 5})
+      const cursors = await get_cursors(ac, events)
+
+      expect(cursors).to.be
+        .equal(["move", "move", "default", "move", "move", "move", "move"])
```

Comment:
I cannot explain why all of the cursors in the array are not "move" - any idea what's going on?

Reply:
The original test suffers similar behavior on occasion, so this isn't unexpected, especially given the overall fragility of cursor handling (specifically in `UIEvents`). I need to do some debugging this get better understanding what's going on on a more fundamental basis.

Reply:
So what next steps should we take? Should I add a comment?

```suggestion
        // TODO: these should all be "move" but there's some glitch that needs to be investigated and fixed
        .equal(["move", "move", "default", "move", "move", "move", "move"])
```

Reply:
I'm pretty sure this isn't going to be a reliable test as-is, because quite likely `"default"` value will appear in various places in this sequence. I will give my self a few more days to dig into this.

Reply:
Got it, thanks!

Reply:
My current understanding is that `cursors` first captures last two hover cursors. Then `default` is due to how pan detection works, when the cursor manages to leave the handle. Finally panning is detected and other `move` cursors belong to pan gesture. In my experiments this test is reliable enough to leave as-is, but improvements will need be done in future, though I'm not exactly sure how to improve things at the moment.